Lib/test/test_asyncgen.py

```diff
@@ -2056,5 +2053,36 @@ async def agenfn():
         del gen2
         gc_collect()  # does not warn unawaited
 
+class TestAsyncGenReturn(unittest.TestCase):
+    def setUp(self):
+        self.loop = asyncio.new_event_loop()
+        asyncio.set_event_loop(None)
+
+    def tearDown(self):
+        self.loop.close()
+        self.loop = None
+        asyncio.set_event_loop_policy(None)
+
+    def test_async_gen_return_value(self):
+        async def gen():
+            yield 1
+            yield 2
+            return 3
```

Comment:
It might be a good idea to add a dummy coroutine (something like `asyncio.sleep(0)` should work) here, as that needs to get yielded in a different way--in my experience with dealing with the coroutine implementation, a coroutine that awaits nothing behaves differently than one that does.

Reply:
Thanks so much for your thorough and prompt review. I've updated this to include awaited calls to `asyncio.sleep`.